File: netlogo-headless/src/main/prim/_countother.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.agent.AgentSet
import org.nlogo.nvm.{ Context, Reporter }

class _countother extends Reporter {

  override def report(context: Context): java.lang.Double =
    Double.box(report_1(context, argEvalAgentSet(context, 0)))

  def report_1(context: Context, arg0: AgentSet): Double = {
    var result = 0
    val iter = arg0.iterator
    while(iter.hasNext)
      if (iter.next() ne context.agent)
        result += 1
    result
  }

}

File: netlogo-gui/src/main/prim/_countother.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.agent.AgentIterator;
import org.nlogo.agent.AgentSet;
import org.nlogo.api.LogoException;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;

public final class _countother extends Reporter {


  @Override
  public Object report(Context context)
      throws LogoException {
    return report_1(context, argEvalAgentSet(context, 0));
  }

  public double report_1(Context context, AgentSet arg0) {
    int result = 0;
    for (AgentIterator iter = arg0.iterator(); iter.hasNext();) {
      if (iter.next() != context.agent) {
        result++;
      }
    }
    return result;
  }
}
